src/main/java/org/spongepowered/api/service/scheduler/TaskBuilder.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.service.scheduler;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Represents a builder to create a {@link Task}.
+ */
+public interface TaskBuilder {
+
+    /**
+     * Sets whether the task should run asynchronous, outside of the main loop,
+     * and in it's own thread. By default, tasks are synchronous.
+     *
+     * <p>A synchronous task is ran in alignment with the game's main loop, in
+     * the same thread. Each synchronous task is ran in series with the tick
+     * cycle. It is safe to manipulate game data when running in this mode.</p>
+     *
+     * <p>In contrast, a task set to run asynchronously will run independently
+     * of the tick cycle and in it's own thread. Therefore the task is <b>not
+     * thread safe</b> with game data and care must be taken. It is strongly
+     * advised to <b>not</b> manipulate game data in asynchronous tasks.</p>
+     *
+     * <p>It is not possible to schedule a task in unit ticks when running
+     * asynchronously, that is why the returned builder is specialised for just
+     * building async tasks. The delay and interval are converted by the
+     * implementation to the equivalent wall clock time.</p>
+     *
+     * @return A specialised asynchronous task builder
```

Comment:
The API doesn't care about the fact the impl may have this as an AsyncTaskBuilder at this point. Docs need to be fixed :).